src/h2n_generate.erl

```diff
@@ -0,0 +1,241 @@
+%%% ---------------------------------------------------------------------------
+%%% @doc
+%%%
+%%%  This module actually generates the final file. It uses John
+%%%  Hughes' brilliant `prettypr` module to output nice, human
+%%%  readable Nix expressions
+%%%  ---------------------------------------------------------------------------
+-module(h2n_generate).
+
+%%
+%% API
+%%
+-export([nix_expression/1]).
+
+-include("hex2nix.hrl").
+%%
+%% Types
+%%
+
+%% We almost never do this, however, with `prettypr` it actually makes
+%% some modicum of sense.
+-import(prettypr, [above/2, beside/2, sep/1, par/1, break/1,
+                   par/2, text_par/1, empty/0]).
+
+%% ============================================================================
+%% Exported Functions
+%% ============================================================================
+-spec nix_expression([h2n_fetcher:dep_desc()]) -> string().
+nix_expression(Deps) ->
```

Comment:
I am not sure which part of the code is responsible, but it would be nice to have different versions of the same package be in the same vicinity in hex-packages.nix, e.g.:

erlcloud_0_11_0 = callPackage ...;
erlcloud_0_12_0 = callPackage ...;
erlcloud = erlcloud_0_12_0;

Currently it seems that all non-newest package versions are moved to the bottom of the file.